### PR TITLE
Persist code changes in condition table when factor update

### PR DESCRIPTION
**Pull request by Yagnik56**
This PR will address issue of code changes not persisting in condition table when factor table get updated as we were generating new table for every edit. Now only rows which are related to code change in factor table will get updated.

Senario: 
If we have 2 factors F1 and F2 and each have 2 levels L1,L2 and L3, L4 
image
then when you change L4 to L5 condition not related L4 persist it's weightage and payload while new condition will be assigned 0 weight.
image
Users can edit values manually as they like or press equal weight to remove weight sum not equal 100 error.
image

### CODE REVIEW

File: frontend/projects/upgrade/src/app/features/dashboard/home/components/factorial-experiment-design/conditions-table/conditions-table.component.ts [hunk 1]
```diff
@@ -105,7 +105,7 @@ export class ConditionsTableComponent implements OnInit, OnDestroy {
     } else {
```
[the remark this answers is not in the document]
Comment (Yagnik56) [reply frontend/projects/upgrade/src/app/features/dashboard/home/components/factorial-experiment-design/conditions-table/conditions-table.component.ts:105]: It won't, I tested it and it work perfectly fine in all scenarios. Please try to check locally, I can explain but you can understand better by debugging yourself.

[the remark this answers is not in the document]
Comment (Yagnik56) [reply frontend/projects/upgrade/src/app/features/dashboard/home/components/factorial-experiment-design/conditions-table/conditions-table.component.ts:106]: Yeah I checked it and it is fixed now, actually I tested it on already exiting exp and forgot try on new on. You can try again, everything seems to be working now.

File: frontend/projects/upgrade/src/app/features/dashboard/home/components/factorial-experiment-design/conditions-table/conditions-table.component.ts [hunk 2]
```diff
@@ -98,15 +98,15 @@ export class ConditionsTableComponent implements OnInit, OnDestroy {
   }
 
   handleDesignDataChanges(designData: ExperimentFactorialDesignData) {
-    if (this.experimentInfo?.partitions.length && !this.formInitialized && !this.isAnyRowRemoved) {
+    const isDesignDataValid = this.checkDesignDataValidity(designData);
+    if (this.experimentInfo?.partitions.length && !this.isAnyRowRemoved && !this.formInitialized) {
       this.handleInitializeExistingTableData();
-    } else if (!this.experimentInfo && this.formInitialized && !this.isAnyRowRemoved) {
+    } else if (!this.experimentInfo && !this.isAnyRowRemoved && (!this.formInitialized || !isDesignDataValid)) {
```
Comment (bcb37) [inline frontend/projects/upgrade/src/app/features/dashboard/home/components/factorial-experiment-design/conditions-table/conditions-table.component.ts:104]: Do we still need `isAnyRowRemoved`? I think `isDesignDataValid` can replace it. And it doesn't go stale after multiple back-and-forth edits.

**Review by bcb37 — COMMENTED** [review]
One more question

File: frontend/projects/upgrade/src/app/features/dashboard/home/components/factorial-experiment-design/conditions-table/conditions-table.component.ts (hunk 2 above)
Comment (Yagnik56) [reply frontend/projects/upgrade/src/app/features/dashboard/home/components/factorial-experiment-design/conditions-table/conditions-table.component.ts]: done

**Review by bcb37 — CHANGES_REQUESTED** [review]
We should remove the places where `isAnyRowRemoved` is set in the `FactorialExperimentDesignComponent` component file, since it is no longer being used in the template.

**Comment by Yagnik56** [comment]
Here you go @bcb37, you can QA and if works fine merge this PR.

**Review by bcb37 — APPROVED** [review]

**Pull request merged**